pkg/cmd/server/origin/run_components.go

```diff
@@ -368,12 +371,83 @@ func (c *MasterConfig) RunDeploymentTriggerController() {
 	dcInfomer := c.Informers.DeploymentConfigs().Informer()
 	rcInformer := c.Informers.ReplicationControllers().Informer()
 	streamInformer := c.Informers.ImageStreams().Informer()
+	// TODO: make this configurable?
```

Comment:
My thought was to move config change triggers in the main controller and remove this controller altogether.